packages/contentprocessing/src/DocumentType.php

```diff
@@ -31,115 +31,195 @@ class DocumentType
-        'application/rar' => 'archive',
-        'application/zip' => 'archive',
-        'application/x-tar' => 'archive',
-        'application/x-bzip2' => 'archive',
-        'application/gzip' => 'archive',
-        'application/x-gzip' => 'archive',
-        'application/x-mimearchive' => 'web-page',
-        'video/x-ms-vob' => 'dvd',
-        'content/DVD' => 'dvd',
-        'video/x-ms-wmv' => 'video',
-        'video/x-ms-wmx' => 'video',
-        'video/x-ms-wm' => 'video',
-        'video/avi' => 'video',
-        'video/divx' => 'video',
-        'video/x-flv' => 'video',
-        'video/quicktime' => 'video',
-        'video/mpeg' => 'video',
-        'video/mp4' => 'video',
-        'video/ogg' => 'video',
-        'video/webm' => 'video',
-        'video/x-matroska' => 'video',
-        'video/3gpp' => 'video',
-        'video/3gpp2' => 'video',
-        'text/csv' => 'spreadsheet',
-        'message/rfc822' => 'email',
-        'application/vnd.ms-outlook' => 'email',
-        'application/gpx+xml' => 'geodata',
-        'application/geo+json' => 'geodata',
-    ];
+        'application/x-mimearchive' => self::WEB_PAGE,
+        
+        'text/uri-list' => self::URI_LIST,
+        
+        'image/jpg' => self::IMAGE,
+        'image/jpeg' => self::IMAGE,
+        'image/gif' => self::IMAGE,
+        'image/png' => self::IMAGE,
+        'image/tiff' => self::IMAGE,
+        'image/svg+xml' => self::IMAGE,
+        'image/vnd.adobe.photoshop' => self::IMAGE,
+        'application/vnd.google-apps.drawing' => self::IMAGE,
+        'application/vnd.oasis.opendocument.graphics' => self::IMAGE,
+
+        'application/rar' => self::ARCHIVE,
+        'application/zip' => self::ARCHIVE,
+        'application/gzip' => self::ARCHIVE,
+        'application/x-tar' => self::ARCHIVE,
+        'application/x-gzip' => self::ARCHIVE,
+        'application/x-bzip2' => self::ARCHIVE,
+        'application/x-7z-compressed' => self::ARCHIVE,
+        
+        'content/DVD' => self::DVD_VIDEO,
+        'video/x-ms-vob' => self::DVD_VIDEO,
+        
+        'video/avi' => self::VIDEO,
+        'video/mp4' => self::VIDEO,
+        'video/ogg' => self::VIDEO,
+        'video/divx' => self::VIDEO,
+        'video/mpeg' => self::VIDEO,
+        'video/webm' => self::VIDEO,
+        'video/3gpp' => self::VIDEO,
+        'video/x-flv' => self::VIDEO,
+        'video/3gpp2' => self::VIDEO,
+        'video/x-ms-wm' => self::VIDEO,
+        'video/x-ms-wmv' => self::VIDEO,
+        'video/x-ms-wmx' => self::VIDEO,
+        'video/quicktime' => self::VIDEO,
+        'video/x-matroska' => self::VIDEO,
+        
+        'message/rfc822' => self::EMAIL,
+        'application/vnd.ms-outlook' => self::EMAIL,
+        
+        'application/gpx+xml' => self::GEODATA,
+        'application/geo+json' => self::GEODATA,
+        'application/vnd.google-earth.kmz' => self::GEODATA,
+        'application/vnd.google-earth.kml+xml' => self::GEODATA,
```

Comment:
Maybe (probably?) I am missing something but it looks like some standard GIS MIME types are missing? Shapefile, Geotiff at least?

I assume that a geotiff should be handled differently then a "normal" tif and that a zip archive containing a shapefile should also be handled differently than a normal archive?

But, again, maybe I am missing something obvious

Reply:
This is just a preparatory work to make the file detection system extensible. Here only the standard mime types are already associated with the respective document type. For files that might be interpreted in various ways there will be a specific pull request with the code to handle that. Since you mentioned shapefile, Geotiff and other GIS related file types you might be interested to have a look at the [Geoserver client](https://github.com/OneOffTech/geoserver-client-php) created by OneOffTech